wsdl-core/src/main/java/io/ballerina/wsdl/core/OperationContext.java

```diff
@@ -0,0 +1,89 @@
+/*
+ *  Copyright (c) 2025, WSO2 LLC. (http://www.wso2.com)
+ *
+ *  WSO2 LLC. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package io.ballerina.wsdl.core;
+
+import static io.ballerina.wsdl.core.Utils.REQUEST_BODY;
+import static io.ballerina.wsdl.core.Utils.RESPONSE_BODY;
+import static io.ballerina.wsdl.core.Utils.SOAP_REQUEST;
+import static io.ballerina.wsdl.core.Utils.SOAP_RESPONSE;
+import static io.ballerina.wsdl.core.WsdlToBallerina.HEADER;
+
+public class OperationContext {
```

Comment:
If this is a data-class, then I would rather use a Java record rather than a class.

[1] - https://www.baeldung.com/java-record-keyword